### PR TITLE
fschat: init at 0.2.28

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

initial stab at fast chat. Tensorflow is broken for now, so waiting to test.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
@natsukium I believe this is ready for review
I have to admin that I looked at the tests and saw open openai and just disabled them.
I'm not 100% sure all the tests use networking.

File: pkgs/development/python-modules/fastchat/default.nix [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+, sentencepiece
+, torch
+, transformers
+, protobuf
+}:
+let
+  version = "0.2.28";
+in
+buildPythonPackage {
+  pname = "fast-chat";
```
Comment (natsukium) [inline pkgs/development/python-modules/fastchat/default.nix:35]:
```suggestion
  pname = "fschat";
```
The pname should be the same as pypi's name.
https://pypi.org/project/fschat/

File: pkgs/development/python-modules/fastchat/default.nix [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      peft
+      sentencepiece
+      torch
+      transformers
+      protobuf
+    ];
+  };
+
+  pythonImportsCheck = [ "fastchat" ];
+  doCheck = false;
```
Comment (natsukium) [inline pkgs/development/python-modules/fastchat/default.nix:92]: It does seem difficult to run the test in a sandbox.
Could you add a comment?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -3707,6 +3707,8 @@ self: super: with self; {
 
   fastcache = callPackage ../development/python-modules/fastcache { };
 
+  fastchat = callPackage ../development/python-modules/fastchat { };
```
Comment (natsukium) [inline pkgs/top-level/python-packages.nix:3710]:
```suggestion
  fschat = callPackage ../development/python-modules/fschat { };
```
I think we should rename an attribute name and a file name, too.

**Review by natsukium — COMMENTED** [review]
besides this, lgtm

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 250918` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.fschat</li>
    <li>python310Packages.fschat.dist</li>
    <li>python311Packages.fschat</li>
    <li>python311Packages.fschat.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**